experimental/form_app/lib/src/sign_in_http.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:flutter/material.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'sign_in_http.g.dart';
+
+@JsonSerializable()
+class FormData {
+  String email;
+  String password;
+
+  FormData({
+    this.email,
+    this.password,
+  });
+
+  factory FormData.fromJson(Map<String, dynamic> json) =>
+      _$FormDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$FormDataToJson(this);
+}
+
+class SignInHttpDemo extends StatefulWidget {
+  final http.Client httpClient;
+
+  SignInHttpDemo({
+    this.httpClient,
+  });
+
+  @override
+  _SignInHttpDemoState createState() => _SignInHttpDemoState();
+}
+
+class _SignInHttpDemoState extends State<SignInHttpDemo> {
+  final _formKey = GlobalKey<FormState>();
```

Comment:
It'd be nice to explain why this is needed with a comment. It's the only thing I found surprising.